lib/src/json_literal_generator.dart

```diff
@@ -28,14 +28,65 @@ class JsonLiteralGenerator extends GeneratorForAnnotation<JsonLiteral> {
         p.join(sourcePathDir, annotation.read('path').stringValue));
     var content = JSON.decode(await buildStep.readAsString(fileId));
 
-    var thing = JSON.encode(content);
+    var asConst = annotation.read('asConst').boolValue;
 
-    var marked = _isConstType(content) ? 'const' : 'final';
+    var thing = _jsonLiteralAsDart(content, asConst).toString();
+    var marked = asConst ? 'const' : 'final';
 
     return '$marked _\$${element.displayName}JsonLiteral = $thing;';
   }
 }
 
-bool _isConstType(value) {
-  return value == null || value is String || value is num || value is bool;
+String _jsonLiteralAsDart(dynamic value, bool asConst) {
```

Comment:
maybe break out some methods?

This is really hard to follow. Also a Doc comment would be nice